internal/osutil/osutil.go

```diff
@@ -210,3 +211,13 @@ func init() {
 func IsWindowsExecutable(path string) bool {
 	return execExts[strings.ToLower(filepath.Ext(path))]
 }
+
+func DiskDetails(path string) (int64, int64, error) {
```

Comment:
I'm not super excited about this name as it's not clear what it returns exactly. It could be clarified by something like

`func DiskUsage(path string) (size, free int64, err error)`

and then have the other one below be `func DiskUsedPercent(path string) float64` or so.